src/frontend/src/components/Posts/Posts.jsx

```diff
@@ -60,7 +72,16 @@ const Posts = () => {
     );
   }
 
-  return (
+  return isDesktopView ? (
```

Comment:
Instead of these long switches, why not make 2 sub components and render one or the other?